vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/DefaultMultiSortPriorityIT.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2000-2022 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.grid.it;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import com.vaadin.flow.component.grid.testbench.GridElement;
+import com.vaadin.flow.component.grid.testbench.GridTHTDElement;
+import com.vaadin.flow.testutil.TestPath;
+import com.vaadin.tests.AbstractComponentIT;
+
+@TestPath("vaadin-grid/default-multi-sort-priority")
+public class DefaultMultiSortPriorityIT extends AbstractComponentIT {
+
+    @Before
+    public void init() {
+        open();
+    }
+
+    @Test
+    public void setDefaultMultiSortPriority_addGrid_priorityIsUsed() {
+        findElement(By.id("btn-set-msp-append")).click();
+        findElement(By.id("btn-add-grid")).click();
```

Comment:
This test actually fails, because the grid is already created before the button click. I think the test passes in CI because it is re-run after a failure, and a second run passes because the static state still applies after the first test run.

I think there are two issues here:
1. The test setup should create the grid instance after changing the setting - this can easily be fixed by creating the grid in the click handler, rather than just add it
2. While that would fix the test setup, it leaves the application in a non-default state, which could have side effects on other tests, so I think the setting should be reverted to its default. One option would be to use a test teardown method, however in this case that would require another Selenium interaction which seems brittle. Instead I would suggest we add a button that does everything in one step: apply the setting, create the grid, then revert the setting. Preferrably with some comment why we revert the setting immediately.

Or we just add a unit test for this, which would be easier to control.

Reply:
> add a button that does everything in one step: apply the setting, create the grid, then revert the setting

Thanks, updated the test.